projects/concurrency/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"container/list"
+	"sync"
+)
+
+type Cache[K comparable, V any] struct {
+	capacity int
+	entries  map[K]*list.Element
+	list     *list.List
+	mu       sync.Mutex
+}
+
+type CacheValue[K comparable, V any] struct {
+	// i don't like that i am duplicating the key here... but it is for easier lookup later
```

Comment:
This seems fine. If you're worried about potentially using a lot of memory, you could keep a pointer to the key here, so you're not making a copy.